cmd/gke-gcloud-auth-plugin/main.go

```diff
@@ -140,6 +180,73 @@ func gcloudConfigOutput() ([]byte, error) {
 	return stdoutBuffer.Bytes(), nil
 }
 
+func cachedGcloudAccessToken(pc *pluginContext) (string, *meta.Time, bool) {
+	token, expiry := pc.cachedTokenEnvVarInput()
+
+	timeStamp, err := time.Parse(time.RFC3339Nano, expiry)
+	if err != nil {
+		klog.V(4).Infof("\nerror parsing time %+v\n\n", err)
+		return "", nil, false
+	}
+
+	tok := &oauth2.Token{
+		AccessToken: token,
+		TokenType:   "Bearer",
+		Expiry:      timeStamp,
+	}
+
+	if !tok.Valid() || tok.Expiry.IsZero() {
+		klog.V(4).Infof("\nerror validating token %+v\n\n", tok)
+		return "", nil, false
+	}
+
+	return tok.AccessToken, &meta.Time{tok.Expiry}, true
+}
+
+func cacheGcloudAccessToken(pc *pluginContext, accessToken string, expiry time.Time) error {
+	po := clientcmd.NewDefaultPathOptions()
+	startingConfig, err := pc.k8sStartingConfig(po)
+	if err != nil {
+		klog.V(4).Infof("Error getting starting config %v", err)
```

Comment:
Why are we using klog in someplaces and fmt.Printf in others?

Reply:
fmt.Printf is only used to print ExecCredential output in the main function without any formatting.

Add klog in case we need debug logs from users.